zerver/management/commands/realm_alias.py

```diff
@@ -32,19 +42,30 @@ def handle(self, *args, **options):
         realm = get_realm(options["string_id"])
         if options["op"] == "show":
             print("Aliases for %s:" % (realm.domain,))
-            for alias in do_get_realm_aliases(realm):
-                print(alias["domain"])
+            for alias in get_realm_aliases(realm):
+                if alias["subdomains_allowed"]:
+                    print("*." + alias["domain"])
+                else:
+                    print(alias["domain"])
             sys.exit(0)
 
-        domain = options['alias'].lower()
+        domain = options['alias'].strip().lower()
         if options["op"] == "add":
-            if not can_add_alias(domain):
+            try:
+                validate_domain(domain)
+                RealmAlias.objects.create(realm=realm, domain=domain, subdomains_allowed=options["subdomains_allowed"])
+                sys.exit(0)
+            except ValidationError as e:
+                print(e.messages[0])
+                sys.exit(1)
+            except DuplicateRealmAliasException:
                 print("A Realm already exists for this domain, cannot add it as an alias for another realm!")
                 sys.exit(1)
-            RealmAlias.objects.create(realm=realm, domain=domain)
-            sys.exit(0)
+            except IntegrityError:
+                print("This domain has already been added!")
+                sys.exit(1)
         elif options["op"] == "remove":
-            RealmAlias.objects.get(realm=realm, domain=domain).delete()
```

Comment:
Even if we change the policy I think it is still going to be confusing in the current form. For example if  there is an alias(realm1, example.com, True) and if the user runs `./manage.py realm_alias --op show --realm 'realm1'` it will be listed as \*.example.com and now suppose he wants to delete it, he would have to enter `./manage.py realm_alias --op remove --realm 'realm1' example.com` which seems a bit confusing. One way of improving it would be to parse the domain entered by the user and if it starts with '*.' then query would be `RealmAlias.objects.get(realm=realm, domain=domain, subdomains_allowed=True)` otherwise `RealmAlias.get(realm=realm, domain=domain, subdomains_allowed=False)`. What do you think?